### PR TITLE
Adding github actions secrets

### PR DESCRIPTION
**Pull request by pdex**
This adds the ability to create repository secrets.

If merged this PR will add a dependency on pynacl.

### CODE REVIEW

**Comment by latacora-tomekr** [comment]
Could we add List Organization Secrets (https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-organization-secrets) to this as well?

**Comment by geerlingguy** [comment]
Tested and works great for my needs! (I just needed to loop through about 100 repositories and add a secret, and it worked for that, as well as updating the secret).

It would be great to support the other methods as well, but I'd love to see this merged as a start as repo secrets are by far the most annoying thing to manage for personal account repositories. At least for orgs, you can add one secret for all the org's repos!

**Comment by pdex** [comment]
@sfdye or @s-t-e-v-e-n-k could you please address this PR? thanks.

**Comment by sfdye** [comment]
will check

**Comment by sfdye** [comment]
@pdex Could you resolve the merge conflicts first?

**Comment by akhmerov** [comment]
What's the status of the PR? The only merge conflict appears to be in requirements / setup.py

I'd be happy to resolve it from a fork if necessary, but I'm not sure if that's the main or only open question.

**Comment by pdex** [comment]
@akhmerov thanks for the bump, I'll get that sorted.

**Comment by pdex** [comment]
@sfdye or @s-t-e-v-e-n-k the merge conflict has been resolved.

Dependency on pynacl has been set to `"pynacl>=1.4.0"` this is now in line with how the other dependencies have been specified.

**Comment by sfdye** [comment]
Thanks for resolving the conflicts. Looks like the lint checks failed, you can fix it by following:

https://github.com/PyGithub/PyGithub/blob/master/CONTRIBUTING.md#coding-style

**Comment by pdex** [comment]
@sfdye all checks are passing.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+pynacl
```
Comment (sfdye) [inline requirements.txt:1]: maybe same here as well
```suggestion
pynacl>=1.4.0
```

**Review by sfdye — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (pdex) [reply requirements.txt]: agreed!

**Comment by pdex** [comment]
@sfdye Permission to merge?

**Comment by sfdye** [comment]
Yes, merge at your will

**Comment by akhmerov** [comment]
Could secrets also be included in a new release after the merge?

**Comment by sfdye** [comment]
Definitely we can cut a new release for secrets

________________________________

**Comment by pdex** [comment]
@sfdye I don't have write permission, could you please merge?

**Pull request merged**